accord-core/src/main/java/accord/messages/WaitOnCommit.java

```diff
@@ -141,13 +147,13 @@ public Void reduce(Void o1, Void o2)
     @Override
     public void accept(Void result, Throwable failure)
     {
-        ack();
+        ack(failure);
     }
 
-    private void ack()
+    private void ack(@Nullable Throwable fail)
     {
         if (waitingOnUpdater.decrementAndGet(this) == -1)
-            node.reply(replyTo, replyContext, WaitOnCommitOk.INSTANCE);
+            node.reply(replyTo, replyContext, fail != null ? WaitOnCommitOk.INSTANCE : null, fail);
```

Comment:
this doesn't look right, if we see a failure will we actually decrement completely?  Also wouldn't the exception be lost if it isn't the last event?

Reply:
Good catch. I will refactor it to check in `accept` and only send the failure response if it hasn't already sent a regular response and log if somehow we have failure but already sent a response.